File: ulohy/11-dva-trojuhelniky/reseni/main.py
import sys


def read_float() -> float:
    return float(input())


def is_valid(a: float, b: float, c: float) -> bool:
    longest_side = max(a, max(b, c))
    return longest_side < a + b + c - longest_side


def are_congruent(a: float, b: float, c: float, k: float, l: float, m: float) -> bool:
    return (a == k and ((b == l and c == m) or (b == m and c == l))) \
        or (a == l and ((b == k and c == m) or (b == m and c == k))) \
        or (a == m and ((b == k and c == l) or (b == l and c == k)))


def are_similar(a: float, b: float, c: float, k: float, l: float, m: float) -> bool:
    ratio = a / k
    if (b / l == ratio and c / m == ratio) or (c / l == ratio and b / m == ratio):
        return True

    ratio = b / k
    if (a / l == ratio and c / m == ratio) or (c / l == ratio and a / m == ratio):
        return True

    ratio = c / k
    if (a / l == ratio and b / m == ratio) or (b / l == ratio and a / m == ratio):
        return True

    return False


def solve(a: float, b: float, c: float, k: float, l: float, m: float) -> str:
    if not is_valid(a, b, c) or not is_valid(k, l, m):
        return "invalid"

    if are_congruent(a, b, c, k, l, m):
        return "congruent"

    if are_similar(a, b, c, k, l, m):
        return "similar"

    return "different"


if __name__ == "__main__":
    a = read_float()
    b = read_float()
    c = read_float()
    k = read_float()
    l = read_float()
    m = read_float()

    print(solve(a, b, c, k, l, m))

File: ulohy/11-dva-trojuhelniky/reseni/main.js
const readline = require("readline");
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
});

let lines = 0;
let a, b, c, k, l, m;

rl.on("line", (line) => {
    lines++;
    switch (lines) {
        case 1:
            a = parseFloat(line);
            return;

        case 2:
            b = parseFloat(line);
            return;

        case 3:
            c = parseFloat(line);
            return;

        case 4:
            k = parseFloat(line);
            return;

        case 5:
            l = parseFloat(line);
            return;

        case 6:
            m = parseFloat(line);
            break;
    }

    console.log(solve(a, b, c, k, l, m));
});

function solve(a, b, c, k, l, m) {
    if (!isValid(a, b, c) || !isValid(k, l, m))
        return "invalid";

    if (areCongruent(a, b, c, k, l, m))
        return "congruent";

    if (areSimilar(a, b, c, k, l, m))
        return "similar";

    return "different";
}

function isValid(a, b, c) {
    var longestSide = Math.max(a, Math.max(b, c));
    return longestSide < a + b + c - longestSide;
}

function areCongruent(a, b, c, k, l, m) {
    return (a === k && ((b === l && c === m) || (b === m && c === l)))
        || (a === l && ((b === k && c === m) || (b === m && c === k)))
        || (a === m && ((b === k && c === l) || (b === l && c === k)));
}

function areSimilar(a, b, c, k, l, m) {
    let ratio = a / k;
    if ((b / l === ratio && c / m === ratio) || (c / l === ratio && b / m === ratio))
        return true;

    ratio = b / k;
    if ((a / l === ratio && c / m === ratio) || (c / l === ratio && a / m === ratio))
        return true;

    ratio = c / k;
    if ((a / l === ratio && b / m === ratio) || (b / l === ratio && a / m === ratio))
        return true;

    return false;
}
